### PR TITLE
feat(logging): Add (*Logger). StandardLoggerFromTemplate() method.

### PR DESCRIPTION
**Pull request by tang-fh**
This method is a generalised but more expensive analog of (*Logger).StandardLogger() which takes a prototype Entry struct as argument.

### CODE REVIEW

**Comment by daniel-sanche** [comment]
This PR would have breaking changes over the last release. We need to either change it to keep existing functionality stable, or save it for [v2.0.0](https://github.com/googleapis/google-cloud-go/milestone/15)

**Comment by tang-fh** [comment]
> This PR would have breaking changes over the last release. We need to either change it to keep existing functionality stable, or save it for [v2.0.0](https://github.com/googleapis/google-cloud-go/milestone/15)

This wasn't intended to be a breaking change. See my issue comment https://github.com/googleapis/google-cloud-go/issues/7207#issuecomment-1387140558 .

File: logging/logging.go [hunk 1]
```diff
@@ -301,16 +302,15 @@ func (c *Client) Logger(logID string, opts ...LoggerOption) *Logger {
...
-	s Severity
+	e *Entry // A prototype Entry object
 }
 
-func (w severityWriter) Write(p []byte) (n int, err error) {
-	w.l.Log(Entry{
-		Severity: w.s,
-		Payload:  string(p),
-	})
+func (w protoEntryWriter) Write(p []byte) (n int, err error) {
```
Comment (daniel-sanche) [inline logging/logging.go:310]: nit: The `proto` prefix usually is typically used to mean `protobuf` elsewhere in the codebase. I think this should be renamed to reduce potential confusion. Maybe call it a `template` instead?

File: logging/logging_test.go [hunk 2]
```diff
@@ -717,6 +717,47 @@ func TestStandardLogger(t *testing.T) {
...
 }
 
+func TestStandardLoggerFromEntry(t *testing.T) {
+	initLogs() // Generate new testLogID
+	ctx := context.Background()
+	lg := client.Logger(testLogID)
+
+	slg := lg.StandardLoggerFromEntry(&logging.Entry{
+		Severity: logging.Info,
+		Trace:    "projects/P/traces/105445aa7843bc8bf206b120001000",
```
Comment (daniel-sanche) [inline logging/logging_test.go:727]: More tests with different metadata filled out would be useful. Especially with payloads in the prototype entry, we want to make sure that's well covered by tests

Comment (tang-fh) [reply logging/logging_test.go]: Done

File: logging/logging.go (hunk 1 above)
Comment (tang-fh) [reply logging/logging.go]: Done.

Good idea. I also think of `proto` to mean proto message. `template` is better.

**Review by daniel-sanche — APPROVED** [review]
LGTM

**Pull request merged**